tests/integrations/raftstore/test_early_apply.rs

```diff
@@ -150,20 +165,23 @@ fn test_update_internal_apply_index() {
     let last_index = cluster.raft_local_state(1, 1).get_last_index();
     cluster.async_remove_peer(1, new_peer(4, 4)).unwrap();
     cluster.async_put(b"k2", b"v2").unwrap();
-    let mut snaps = vec![];
-    for i in 1..3 {
-        cluster.wait_last_index(1, i, last_index + 2, Duration::from_secs(3));
-        snaps.push((i, RocksSnapshot::new(cluster.get_raft_engine(1))));
+    let mut snaps = Vec::new();
+    for id in 1..3 {
+        cluster.wait_last_index(1, id, last_index + 2, Duration::from_secs(3));
+        snaps.push((id, dump_raft_entries(&cluster.get_raft_engine(id), id)));
     }
     cluster.clear_send_filters();
     must_get_equal(&cluster.get_engine(1), b"k2", b"v2");
     must_get_equal(&cluster.get_engine(2), b"k2", b"v2");
 
     // Simulate data lost in raft cf.
-    for (id, snap) in &snaps {
-        cluster.stop_node(*id);
-        cluster.restore_raft(1, *id, snap);
-        cluster.run_node(*id).unwrap();
+    for (id, mut batch) in snaps.into_iter() {
```

Comment:
```suggestion
    for (id, mut batch) in snaps {
```

Reply:
The batch are drained, I think into is better.

Reply:
Actually, `for a in vec` works like `let iter = vec.into_iter(); while let Some(i) = iter.next()`. For loop only accepts types implements `IntoIterator` trait.

Reply:
Oh, right.